apiserver/facades/client/application/application.go

```diff
@@ -1930,13 +1892,34 @@ func (api *APIBase) SetConfigs(ctx context.Context, args params.ConfigSetArgs) (
 	}
 	result.Results = make([]params.ErrorResult, len(args.Args))
 	for i, arg := range args.Args {
-		app, err := api.backend.Application(arg.ApplicationName)
-		if err != nil {
+		if arg.ConfigYAML != "" {
+			result.Results[i].Error = apiservererrors.ServerError(errors.NotImplementedf("config yaml not supported"))
+			continue
+		}
+
+		appID, err := api.applicationService.GetApplicationIDByName(ctx, arg.ApplicationName)
+		if errors.Is(err, applicationerrors.ApplicationNotFound) {
+			result.Results[i].Error = apiservererrors.ServerError(errors.NotFoundf("application %q", arg.ApplicationName))
+			continue
+		} else if errors.Is(err, applicationerrors.ApplicationNameNotValid) {
+			result.Results[i].Error = apiservererrors.ServerError(errors.NotValidf("application name %q", arg.ApplicationName))
+			continue
+		} else if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+
+		err = api.applicationService.UpdateApplicationConfig(ctx, appID, arg.Config)
+		if errors.Is(err, applicationerrors.ApplicationNotFound) {
+			result.Results[i].Error = apiservererrors.ServerError(errors.NotFoundf("application %q", arg.ApplicationName))
+			continue
+		} else if errors.Is(err, applicationerrors.InvalidApplicationConfig) {
+			result.Results[i].Error = apiservererrors.ServerError(internalerrors.Errorf("%w%w", err, errors.Hide(errors.NotValid)))
+			continue
+		} else if err != nil {
 			result.Results[i].Error = apiservererrors.ServerError(err)
 			continue
 		}
```

Comment:
We could clean this up by calling a method that returns an error. That way, we don't need all the continues.

Reply:
Sure :+1: